pytest_splunk_addon/standard_lib/utilities/create_new_eventgen.py

```diff
@@ -0,0 +1,245 @@
+import argparse
+import re
+import os
+
+from splunk_appinspect import App
+from .mapping import FIELD_MAPPING, FILE_MAPPING
+
+
+class UpdateEventgen():
+    """ Update eventgen file """
+
+    def __init__(self, addon_path):
+        self._app = App(addon_path, python_analyzer_enable=False)
+        self._eventgen = None
+        self.path_to_samples = os.path.join(addon_path, "samples")
+
+    @property
+    def eventgen(self):
+        try:
+            if not self._eventgen:
+                self._eventgen = self._app.get_config("eventgen.conf")
+            return self._eventgen
+        except OSError:
+            raise Exception("Eventgen.conf not found")
+            return None
+
+    def get_eventgen_stanzas(self):
+        """
+        To get eventgen stanza and create a dictionary. 
+        If stanza contains regex for multiple sample files, then it creates stanza for each sample file.
+
+        Return: 
+            eventgen_dict (dict):
+                {
+                    "stanza_name": {    
+                        "other metadata": "source, sourcetype, etc."
+                        "expected_event_count" : int
+                        "tokens": {
+                            0: {
+                                token: #One#
+                                replacementType: random
+                                replacement: static 
+                            }
+                        }
+                    }
+                }
+        """
+        eventgen_dict = {}
+        for stanza in self.eventgen.sects:
+            # events_in_file = 0
+            eventgen_sections = self.eventgen.sects[stanza]
+            eventgen_dict.setdefault((stanza), {
+                'tokens': {},
+            })
+
+            for stanza_param in eventgen_sections.options:
+                eventgen_property = eventgen_sections.options[stanza_param]
+                if eventgen_property.name.startswith('token'):
+                    _, token_id, token_param = eventgen_property.name.split(
+                        '.')
+                    if not token_id in eventgen_dict[stanza]['tokens'].keys():
+                        eventgen_dict[stanza]['tokens'][token_id] = {
+                        }
+                    eventgen_dict[stanza]['tokens'][token_id][token_param] = eventgen_property.value
+                    try:
+                        events_in_file = len(open(os.path.join(self.path_to_samples, stanza)).readlines())
```

Comment:
Why are we reading files for all the param in the stanza? Can we not do the same thing before this for loop (`for stanza_param in eventgen_sections.options`)?

Reply:
Yes, it can be done. Will update it.